### PR TITLE
Create react-native-ci.yml

### PR DESCRIPTION
**Pull request by maryjabro**
Added GitHub Actions workflow for Android APK build and testing. Needs reviewal to ensure correctness

### CODE REVIEW

File: .github/workflows/react-native-ci.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+name: React Native Frontend CI
+
+
+on:
+  push:
+    branches:
+      - MaryJabro  # Temporary for testing: Adjust to main before pushing to main
```
Comment (MerwinAamir1) [inline .github/workflows/react-native-ci.yml:7]: oops looks like you missed this. need to change this to main

File: .github/workflows/react-native-ci.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  pull_request:
+    branches:
+      - MaryJabro  #Temporary for testing:  Adjust to main before pushing to main
```
Comment (MerwinAamir1) [inline .github/workflows/react-native-ci.yml:10]: Same as above

**Review by MerwinAamir1 — COMMENTED** [review]
Small mistake here

**Comment by MerwinAamir1** [comment]
Another thing is when this eventually gets approved lets make sure we squash and merge since you have 11 commits. The reason we want to do this is because we don't want to pollute the main branch with a bunch of commits that don't make much sense. Also if something goes wrong we can pinpoint the commit that caused it easier.

File: .github/workflows/react-native-ci.yml (hunk 1 above)
Comment (maryjabro) [reply .github/workflows/react-native-ci.yml:7]: > oops looks like you missed this. need to change this to main

updating it to main right now, had it as MaryJabro for testing

**Comment by MerwinAamir1** [comment]
Looks good to me now. Approved

**Pull request merged**